cmd/store.go

```diff
@@ -23,52 +24,46 @@ func (s *store) Get(IP string) (string, error) {
 	if role, ok := s.rolesByIP[IP]; ok {
 		return role, nil
 	}
+	log.Warnf("Using fallback role for IP %s", IP)
 	if s.defaultRole != "" {
-		log.Warnf("Using fallback role for IP %s", IP)
 		return s.defaultRole, nil
 	}
 	return "", fmt.Errorf("Unable to find role for IP %s", IP)
 }
 
 // OnAdd is called when a pod is added.
 func (s *store) OnAdd(obj interface{}) {
-	if pod, ok := obj.(*api.Pod); ok {
-		if pod.Status.PodIP != "" {
-			if role, ok := pod.Annotations[s.iamRoleKey]; ok {
-				s.mutex.Lock()
-				s.rolesByIP[pod.Status.PodIP] = role
-				s.mutex.Unlock()
-			}
+	pod := obj.(*api.Pod)
```

Comment:
The interface is generic, but we explicitly only watch for pods: https://github.com/jtblin/kube2iam/blob/master/cmd/k8s.go#L31 .

I'll make a change to log errors instead of crashing, but I think I'll keep running it with the crash code. The `Informer` cache is a big unknown to me and this `DeletedFinalStateUnknown` is probably just the tip of the iceberg. This value doesn't come from kubernetes itself - it's part of the informer framework (i.e. the API). Crashing is definitely safer from a security perspective and I'd like to keep that in my cluster.